ICAFIX/ReApplyFixMultiRunPipeline.sh

```diff
@@ -168,11 +173,63 @@ determine_old_or_new_fsl()
 have_hand_reclassification()
 {
 	local StudyFolder="${1}"
-	local Subject="${2}"
+	local Session="${2}"
 	local fMRIName="${3}"
 	local HighPass="${4}"
 
-	[ -e "${StudyFolder}/${Subject}/MNINonLinear/Results/${fMRIName}/${fMRIName}_hp${HighPass}.ica/HandNoise.txt" ]
+	[ -e "${StudyFolder}/${Session}/MNINonLinear/Results/${fMRIName}/${fMRIName}_hp${HighPass}.ica/HandNoise.txt" ]
+}
+
+# ------------------------------------------------------------------------------
+# Copy necessary files from cross-sectional to longitudinal target
+# ------------------------------------------------------------------------------
+function copy_to_longitudinal()
+{
+	local StudyFolder="$1" SessionCross="$2" SessionLong="$3" fMRINames="$4" ConcatName="$5" HighPass="$6"
+	local fmri S T
+	#copy for concatentated run
+	S="$StudyFolder/$SessionCross/MNINonLinear/Results/$ConcatName"
+	T="$StudyFolder/$SessionLong/MNINonLinear/Results/$ConcatName"
+	local file files_to_copy="Movement_Regressors_demean.txt ReclassifyAsNoise.txt ReclassifyAsSignal.txt"
+	mkdir -p $T
+	for file in $files_to_copy; do
+		cp $S/$file $T/
+	done
+	
+	local ICADir="$S/${ConcatName}_hp$HighPass.ica"
+	local ICADirLong="$T/${ConcatName}_hp$HighPass.ica"
+	mkdir -p "$ICADirLong"				
+	if [ -f "$ICADir/HandNoise.txt" ]; then 
+		cp $ICADir/HandNoise.txt $ICADirLong/
+	fi
+	files_to_copy="fix4melview_HCP_Style_Single_Multirun_Dedrift_thr10.txt fix4melview_HCP_Style_Single_Multirun_Dedrift_thr10.wb_annsub.csv hand_labels_noise.txt HandNoise.txt \
+	HandSignal.txt Noise.txt ReclassifyAsNoise.txt ReclassifyAsSignal.txt Signal.txt .fix"
+	for file in $files_to_copy; do
+		cp $ICADir/$file $ICADirLong/
+	done 
+	
+	mkdir -p $ICADirLong/mc
+	cp $ICADir/mc/prefiltered_func_data_mcf_conf.nii.gz $ICADirLong/mc/
+	cp $ICADir/mc/prefiltered_func_data_mcf_conf_hp.nii.gz $ICADirLong/mc/
+	cp $ICADir/mc/prefiltered_func_data_mcf.par $ICADirLong/mc/
+	
+	mkdir -p $ICADirLong/fix/
+	cp $ICADir/fix/features.csv $ICADirLong/fix/
+	
+	files_to_copy="eigenvalues_percent ICAVolumeSpace.txt melodic_FTmix melodic_FTmix.sdseries.nii \
+	melodic_ICstats melodic_mix melodic_mix.sdseries.nii melodic_Tmodes melodic_unmix"
+	mkdir -p $ICADirLong/filtered_func_data.ica
+	for file in $files_to_copy; do
+		cp $ICADir/filtered_func_data.ica/$file $ICADirLong/filtered_func_data.ica/
+	done	
+	
+	#copy for individual fMRI runs
+	for fmri in $fMRINames; do		
+		S="$StudyFolder/$SessionCross/MNINonLinear/Results/$fmri"
+		T="$StudyFolder/$SessionLong/MNINonLinear/Results/$fmri"
+		mkdir -p $T/${fmri}_hp$HighPass.ica/mc
+		cp $S/${fmri}_hp$HighPass.ica/mc/prefiltered_func_data_mcf.par $T/${fmri}_hp$HighPass.ica/mc/		
```

Comment:
Underscores outside of quotes will be literal, so you can just quote each variable without using `{}`.
```suggestion
		mkdir -p "$T"/"$fmri"_hp"$HighPass".ica/mc
		cp "$S"/"$fmri"_hp"$HighPass".ica/mc/prefiltered_func_data_mcf.par "$T"/"$fmri"_hp"$HighPass".ica/mc/		
```